lib/src/vector_math/matrix4.dart

```diff
@@ -200,6 +200,29 @@ class Matrix4 {
         ..setValues(arg0, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9,
             arg10, arg11, arg12, arg13, arg14, arg15);
 
+  /// New matrix only with the specified values.
+  factory Matrix4.only({
+    double arg0 = 0,
+    double arg1 = 0,
+    double arg2 = 0,
+    double arg3 = 0,
+    double arg4 = 0,
+    double arg5 = 0,
+    double arg6 = 0,
```

Comment:
I think these arguments need better names.
Is `Matrix4.only(arg4: 2.1)` setting an element on the diagonal? It is too hard to tell, and is completely different answer to `Matrix3.only(arg4: 2.1)`.

The vector names are nice.

I do question the need for these constructors since we have cascades which already provides a succinct way of specifying just a few elements of the matrix or vector. Does writing / testing / maintaining this close alternative really pay its way?

One can write the following with cascades
```
var v = Vector4.zero()..y = 2.1;
var m = Matrix4.zero()..setEntry(1, 0, 2.1);
```
It is only a slight improvement to write
```
var v = Vector4.only(y: 2.1);
var m = Matrix4.only(arg4: 2.1);
```
The matrix version with cascades would be tighter if we had getters/setters for each position, e.g. `Matrix4.zero()..m10 = 2.1` or `Matrix4.zero()..r1c0 = 2.1`.

Optional arguments do come at a runtime cost to figure out which values were passed and which should use the default value, a cost not present in the cascade style, and writers of math code often are concerned with performance.

Reply:
I think that in the "`setEntry`-way" it is much clearer which field of the matrix that is actually changed, compared to the suggested Matrix constructors. Maybe it is possible to somehow give the constructor arguments clearer names though, but if you are adding more than a few values with the named arguments in the constructor I think you'd want to write out the whole matrix anyways (most of the time) to improve readability.

Reply:
> I think these arguments need better names.
Is Matrix4.only(arg4: 2.1) setting an element on the diagonal? It is too hard to tell, and is completely different answer to Matrix3.only(arg4: 2.1).

I definitely agree, they deserve better naming. If we want to keep the `only` constructor I'm happy renaming them to something such as, for example, `r1c2`.

> Optional arguments do come at a runtime cost to figure out which values were passed and which should use the default value, a cost not present in the cascade style, and writers of math code often are concerned with performance.

I definitely agree with the above. I'm quite happy with the proposed `Matrix4.zero()..r1c0 = 2.1` for matrices. If we are all happy with it (cc : @spydon , @rakudrama ) I can implement these setters. 

---

As you pointed out `var v = Vector4.only(y: 2.1);` has a slight improvement over `var v = Vector4.zero()..y = 2.1;`. If we think we should always encourage the library to provide the best performant API, then we should get rid of `.only` and implement getters. 

On some occasions, due to formatting, the `.only` has a "slight improvement", I'm happy to sacrifice being less verbose for performance.

Reply:
For the naming convention for the Matrices I would suggest something that we did on [Flame](https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/extensions/matrix4.dart#L7) for the `Matrix4`.

This naming convention is quite common in game engines and (at least here in the Netherlands) in math calculations.  There are other variants of it that would also work. But the most important part is that it should reflect what row/column on a matrix u are accessing.

Reply:
@wolfenrain I'm happy with `m11` over `r1c1`. I think ideally it would be nice to remove the need for [this](https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/extensions/matrix4.dart#L7) living in Flame.

@spydon I replied over [here](https://github.com/google/vector_math.dart/pull/266#discussion_r892265325).